arrow/src/main/java/org/apache/calcite/adapter/arrow/ArrowTable.java

```diff
@@ -122,7 +122,13 @@ public Enumerable<Object> query(DataContext root, ImmutableIntList fields,
         treeNodes.add(
             TreeBuilder.makeField(schema.getFields()
                 .get(schema.getFields().indexOf(schema.findField(data[0])))));
-        treeNodes.add(makeLiteralNode(data[2], data[3]));
+
+        // if the split condition has more than two parts it's a binary operator
+        // with an additional literal node
+        if (data.length > 2) {
+          treeNodes.add(makeLiteralNode(data[2], data[3]));
+        }
+
         String equality = data[1];
```

Comment:
This is not necessarily an equality, right?

Reply:
Nope, it isn't. Gandiva refers to it as a function, but I think `operator` definitely makes way more sense. Adjusted with [Rename equality to operator in ArrowTable](https://github.com/apache/calcite/pull/3858/commits/a7c195b43cc0e7b1d6c84bcdd1873efebb3be0fb)